src/components/feedback/Feedback.tsx

```diff
@@ -1,11 +1,152 @@
-import React, { FC } from 'react';
+import React, { FC, useReducer, ChangeEvent, useMemo, useCallback, MouseEvent } from 'react';
 
 import commonStyle from '../../styles/commonStyles.css';
+import {
+  feedbackReducer,
+  CHANGE_FEEDBACK_STATE,
+  RESET_STATE
+} from '../../reducers/feedbackReducer';
+import styles from './Feedback.css';
+import useValidate from 'hooks/useValidate';
+import { setInLocalStorage, FEEDBACK_DATA_LS_LABEL } from 'services/localStorage';
 
 export const Feedback: FC = () => {
+  const [state, dispatch] = useReducer(feedbackReducer, {
+    name: '',
+    email: '',
+    problems: '',
+    rating: 5
+  });
+
+  const email = useValidate();
+  const name = useValidate();
+
+  const stateHandler = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    dispatch({
+      type: CHANGE_FEEDBACK_STATE,
+      payload: {
+        name: e.target.name,
+        data: e.target.type === 'range' ? +e.target.value : e.target.value
+      }
+    });
+  };
+
+  const requiredInputHandler = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.name === 'name') {
+      name.validateInput(e.target, e.target.value);
+    }
+    if (e.target.name === 'email') {
+      email.validateInput(e.target, e.target.value);
+    }
```

Comment:
conditions can be refactored. in any case you'll call `validateInput` method with the same params, using condition you can decide what control to select

Reply:
but I need to call different `validateInput`. It depends on conditions

Reply:
as an option:
```
let selectedInput;
if (...) {
  selectedInput = ...
}
selectedInput?.validateInput(e.target, e.target.value);
```